impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -35,6 +35,358 @@ define(["./_focus"], function(focusUtil) {
+            nextItem = findNextOption(currentItem);
+          }
+
+          if (nextItem && event.shiftKey) {
+            selectRange(startRangeIndex, nextItem);
+          }
+
+          if (nextItem) {
+            setActiveOption(nextItem);
+            evt.preventDefault();
+          }
+
+          break;
+        case keyCodes.home:
+          evt.preventDefault();
+          focusFirstItem();
+
+          if (evt.shiftKey && evt.ctrlKey) {
+            selectRange(startRangeIndex, 0);
+          }
+          break;
+        case keyCodes.end:
+          evt.preventDefault();
+          focusLastItem();
+
+          if (evt.shiftKey && evt.ctrlKey) {
+            selectRange(startRangeIndex, allOptions.length - 1);
+          }
+          break;
+        case keyCodes.shift:
+          startRangeIndex = getOptionIndex(currentItem);
+          break;
+        case keyCodes.space:
+          evt.preventDefault();
+          toggleSelectItem(nextItem);
+          break;
+        case 65:
+          // handle control + A
+          if (evt.ctrlKey || evt.metaKey) {
+            evt.preventDefault();
+            selectRange(0, allOptions.length - 1);
+            break;
+          }
+          // fall through for Type-ahead
+        default:
+          var character = String.fromCharCode(key).toUpperCase();
+          var startIndex = !keysSoFar ? getActiveOptionIndex() : null;
+
+          keysSoFar += character;
+
+          clearKeysSoFarAfterDelay();
+
+          var optionToFocus = findOption(keysSoFar, startIndex);
+          if (optionToFocus) {
+            setActiveOption(optionToFocus);
+          }
+          break;
+      }
+
+      if (activeDescendantId !== lastActiveId) {
+        updateScroll();
+      }
+    }
+
+    /**
+     * Helper function to check if a number is within a range; no side effects.
+     *
+     * @param index
+     * @param start
+     * @param end
+     * @returns {boolean}
+     */
+    function checkInRange(index, start, end) {
+      var rangeStart = start < end ? start : end;
+      var rangeEnd = start < end ? end : start;
+
+      return index >= rangeStart && index <= rangeEnd;
+    }
+
+    /**
+     * Select a range of options
+     *
+     * @param start
+     * @param end
+     */
+    function selectRange(start, end) {
+      // get start/end indices
+      var allOptions = getAllOptionsArray();
+      var startIndex = typeof start === "number" ? start : allOptions.indexOf(start);
+      var endIndex = typeof end === "number" ? end : allOptions.indexOf(end);
+
+      for (var index = 0; index < allOptions.length; index++) {
+        var selected = checkInRange(index, startIndex, endIndex);
+        allOptions[index].setAttribute("aria-selected", selected + "");
+        if (selected) {
+          allOptions[index].classList.add("selected");
+        } else {
+          allOptions[index].classList.remove("selected");
+        }
+      }
+    }
+
+    /**
+     * Focus on the first option
```

Comment:
Text and punctuation: `Sets the first option as the active option.`.
Regarding punctuation, rinse and repeat.